test/norm_num_ext.lean

```diff
@@ -243,6 +243,13 @@ example : squarefree 10 := by norm_num
 example : squarefree (2*3*5*17) := by norm_num
 example : ¬ squarefree (2*3*5*5*17) := by norm_num
 example : squarefree 251 := by norm_num
+example : squarefree (3 : ℤ) :=
+begin
+  -- `norm_num` should fail on this example, instead of producing an incorrect proof.
+  success_if_fail { norm_num },
+  exact irreducible.squarefree (prime.irreducible
+    (int.prime_iff_nat_abs_prime.mpr (by norm_num)))
+end
 
```

Comment:
Can you add this testcase too?
```lean
example : @squarefree ℕ multiplicative.monoid 1 :=
begin
  norm_num
end
```

Reply:
Here's the proof that it's true:
```lean
example : @squarefree ℕ multiplicative.monoid 1 :=
begin
  -- the statement was deliberately wacky, let's fix it
  change squarefree (multiplicative.of_add 1 : multiplicative ℕ),
  rintros x ⟨dx, hd⟩,
  revert x dx,
  rw multiplicative.of_add.surjective.forall₂,
  intros x dx h,
  simp_rw [←of_add_add, multiplicative.of_add.injective.eq_iff] at h,
  cases x,
  { simp [is_unit_one], exact is_unit_one },
  { simp only [nat.succ_add, nat.add_succ] at h,
    cases h },
end
```

Obviously I don't expect `norm_num` to prove it, but I would expect it not to emit an invalid proof.

Reply:
Looks like `norm_num` fails on this example (as it should). I'll add the test case to cover the case of the different monoid instance anyway.